src/algorithms/contractions/ctmrg_contractions.jl

```diff
@@ -350,6 +350,234 @@ function half_infinite_environment(
         conj(E_4[χ6 D9 D10; χ_in])
 end
 
+"""
+    full_infinite_environment(
+        quadrant1::T, quadrant2::T, quadrant3::T, quadrant4::T
+    ) where {T<:AbstractTensorMap{<:ElementarySpace,3,3}}
+    function full_infinite_environment(
+        half1::T, half2::T
+    ) where {T<:AbstractTensorMap{<:ElementarySpace,3,3}}
+    full_infinite_environment(C_1, C_2, C_3, C_4, E_1, E_2, E_3, E_4, E_5, E_6, E_7, E_8,
+                              ket_1::P, ket_2::P, bra_1::P=ket_1, bra_2::P=ket_2) where {P<:PEPSTensor}
+    full_infinite_environment(C_1, C_2, E_1, E_2, E_3, E_4, x,
+                              ket_1::P, ket_2::P, ket_3::P, ket_4::P,
+                              bra_1::P=ket_1, bra_2::P=ket_2, bra_3::P=ket_3, bra_4::P=ket_4) where {P<:PEPSTensor}
+    full_infinite_environment(x, C_1, C_2, E_1, E_2, E_3, E_4,
+                              ket_1::P, ket_2::P, ket_3::P, ket_4::P,
+                              bra_1::P=ket_1, bra_2::P=ket_2, bra_3::P=ket_3, bra_4::P=ket_4) where {P<:PEPSTensor}
+
+Contract four quadrants (enlarged corners) to form a full-infinite environment.
+
+```
```

Comment:
Well, the diagrams indeed would suggest that you find the projectors for the right move. But in fact the contractions are the correct contractions for any move because only the order of the supplied `quadrant`s (or corner/edge/PEPS tensors) matters.

My point of choosing the diagrams like that was that the first quadrant starts in the northwest corner because that's where our counting starts anyway. But of course the orientation is arbitrary and only the fact that the incoming indices of `quadrant1` and the outgoing indices of `quadrant4` are open is relevant.